### ISSUE TITLE
Errors not being flagged inside a StyledComponent's `switch` statements

### ISSUE THREAD

**Issue by emmasax** [stylelint/stylelint#6468]
The following code will pass linting:

```js
export const Dividers = styled.div`
  ${({ orientation }) => {
    switch (orientation) {
      case 'vertical':
        return css`
          color: #ssssss;
          width: 1px;
          height: 100%;
        `;
      default:
        return css`
          color: #llllll;
          width: 100%;
          height: 1px;
        `;
    }
  }}
`;
```

However, moving the switch statement outside of the StyledComponent, will flag the invalid hex colours:

```js
const handleOrientationStyles = (orientation) => {
  switch (orientation) {
    case 'vertical':
      return css`
        color: #ssssss;
        width: 1px;
        height: 100%;
      `;
    default:
      return css`
        color: #llllll;
        width: 100%;
        height: 1px;
      `;
  }
};

export const Dividers = styled.div`
  ${({ orientation }) => handleOrientationStyles(orientation)}
`;
```

```json
{
  "processors": ["stylelint-processor-styled-components"],
  "extends": ["stylelint-config-styled-components"],
  "customSyntax": "postcss-scss"
}
```

**Comment by ybiquitous** [comment]
@emmasax Thank you for creating this issue. However, issues need to follow one of our templates so that we can clearly understand your particular circumstances.

Please help us help you by [recreating the issue](https://github.com/stylelint/stylelint/issues/new/choose) using one of our templates.

And this problem might be related to `stylelint-config-styled-components` or `stylelint-processor-styled-components` instead of Stylelint itself. Please note that we've discussed the deprecation for CSS-in-JS syntaxes in <https://github.com/stylelint/postcss-css-in-js/issues/225>.
